### PR TITLE
Feat/chat/navigation

### PR DESCRIPTION
**Pull request by taghizadlaura**
# Chat - Create navigation to ChatScreen

## Description
This PR introduces a new navigation button in the top app bar of `AlertListScreen` that allows users to navigate to the `ChatScreen`. It also includes a "Go Back" button in the `ChatScreen`'s top app bar for navigation purposes. This PR closes issue #279 and is part of #275.

Note: I did not implement the logic to prevent having both the Chat button and another button simultaneously, as this is a temporary solution and will be addressed in a future PR.

## Changes
- Implemented a Chat button in the top app bar of `AlertListScreen` and its navigation logic.
- Integrated the navigation action to redirect to `ChatScreen`.
- Added a "Go Back" button in the top app bar of `ChatScreen` to return to the `AlertListScreen`.
- Added an empty `ChatViewModel`.
- Updated tests and test tags to include verification for the new buttons.
- Completed navigation actions and main activity modifications as needed to support navigation between `AlertListScreen` and `ChatScreen`.

#### Added
- `ChatScreen.kt`: UI implementation for the new chat screen.
- `ChatViewModel.kt`: Empty ViewModel for the `ChatScreen`.

#### Modified
- `MainActivity.kt`: Updated for navigation purposes.
- `C.kt`: Added test tag for the chat button.
- `NavigationActions.kt`: Added navigation action to `ChatScreen` and logic for the "Go Back" button.
- `TopAppBar.kt`: Added the new Chat button and "Go Back" button.
- `AlertListScreen.kt`: Integrated the Chat button into the top app bar.
- `TopAppBarTest.kt`: Adapted and completed tests for the Chat and "Go Back" buttons.
- `AlertListScreenTest.kt`: Adapted and completed tests to verify the new navigation button.

#### Removed
- None.

## Dependencies Added
- None.

## Testing
- Adapted and completed tests in `TopAppBarTest.kt` and `AlertListScreenTest.kt` to verify the new Chat and "Go Back" button functionalities.
- End-to-end (E2E) tests passed successfully for navigation to and from the `ChatScreen`.

### CODE REVIEW

**Comment by taghizadlaura** [comment]
I noticed that the line coverage is quite low, and it's mainly due to the `ChatScreen`. This is expected, as I plan to write tests for it in the same PR where I fully implement it. Is this acceptable, or should I add some tests for it now?

File: app/src/main/java/com/android/periodpals/ui/navigation/TopAppBar.kt [hunk 1]
```diff
@@ -66,6 +67,8 @@ fun TopAppBar(
     onSettingsButtonClick: (() -> Unit)? = null,
     editButton: Boolean = false,
     onEditButtonClick: (() -> Unit)? = null,
+    chatButton: Boolean = false,
+    onChatButtonClick: (() -> Unit)? = null,
```
Comment (agonzalez-r) [inline app/src/main/java/com/android/periodpals/ui/navigation/TopAppBar.kt:71]: you didn't add a `require` function to ensure that if `chatButton` is true, there must be a function passed for `onChatButtonClick` (it must be non null)

Comment (agonzalez-r) [inline app/src/main/java/com/android/periodpals/ui/navigation/TopAppBar.kt:71]: You could also add documentation for these, if you find it relevant

**Review by agonzalez-r — APPROVED** [review]
LGTM!
As you mentioned, the chat button in the top app bar is a temporary solution, so I don't think there is any need to do further testing for a feature that will be removed. Also documentation is missing in `TopAppBar` but I guess the importance of adding it depends on how long you plan on keeping this button. 

Adding more things also implies deleting more things on the future so I think it's up to you to find the right balance.

## Review 
- I left a nitpick about adding a `require()` in `TopAppBar`, for consistency with the other buttons, but mostly to be sure every time there is a chat button, there is a callback function to indicate what proceeds when clicked
- You could also add a test for this by ensuring that the following parameters throw an exception: 
```
chatButton = true,
onChatButtonClick = null,
```

**Review by lazarinibruno — APPROVED** [review]
LGTM!

As you mentioned, this is a temporary solution to be able to access the chat 🐈 and thus testing must not be thorough.

Excited to see how the chat turns out! Keep up the good work :)

**Pull request merged**